### DISCUSSION TITLE
API Hook for drowned guest

### DISCUSSION THREAD

**Discussion by Crazycolbster** [OpenRCT2/OpenRCT2#20420] [Feature requests]
I'm currently writing a plugin to make OpenRCT2 compatible with the Archipelago multi-world randomizer system. In there, I'd like to have a hook I can subscribe to to punish the player if a guest or staff member drowns, activating the "Death Link" function. When one player dies, everybody dies, if the options enabled. I'd appreciate this. Thanks!

**Comment by tubbo** [comment]
The [`GuestAnimation`](https://github.com/OpenRCT2/OpenRCT2/blob/develop/distribution/openrct2.d.ts#L2753) type has a `"drowning"` entry, and [`Guest.animation`](https://github.com/OpenRCT2/OpenRCT2/blob/develop/distribution/openrct2.d.ts#L2932) should return that whenever you ask for a `Guest`. I would wager you can check on each tick whether any guest is exhibiting this animation, and perform operations accordingly.

**Reply by Lastorder-DC** [reply]
If guest or staff accidently falls into water, and saved right after that, is animation check return invalid result?(as writer wants drowned check, not drowning check)

(edit) adding general guest/staff delete hook might be useful(if not exist)
(returns why guest/staff disappeared like 'sacked', 'drowned', 'left')

**Reply by Basssiiie** [reply]
I think a hook would be a better. Checking all guests every tick is a very computationally expensive operation, especially if you have a lot of guests.
